src/jit/ssarenamestate.cpp

```diff
@@ -33,42 +33,43 @@ void SsaRenameState::EnsureStacks()
     }
 }
 
-/**
- * Returns a SSA count number for a local variable from top of the stack.
- *
- * @params lclNum The local variable def for which a count has to be returned.
- * @return the current variable name for the "use".
- *
- * @remarks If the stack is empty, then we have an use before a def. To handle this
- *          special case, we need to initialize the count with 'default+1', so the
- *          next definition will always use 'default+1' but return 'default' for
- *          all uses until a definition.
- *
- */
-unsigned SsaRenameState::CountForUse(unsigned lclNum)
+//------------------------------------------------------------------------
+// Top: Get the SSA number at the top of the stack for the specified variable.
+//
+// Arguments:
+//    lclNum - The local variable number
+//
+// Return Value:
+//    The SSA number.
+//
+// Notes:
+//    The stack must not be empty. Method parameters and local variables that are live in at
+//    the start of the first block must have associated SSA definitions and their SSA numbers
+//    must have been pushed first.
+//
+unsigned SsaRenameState::Top(unsigned lclNum)
 {
-    EnsureStacks();
-    DBG_SSA_JITDUMP("[SsaRenameState::CountForUse] V%02u\n", lclNum);
+    DBG_SSA_JITDUMP("[SsaRenameState::Top] V%02u\n", lclNum);
 
+    noway_assert(stacks != nullptr);
     Stack* stack = stacks[lclNum];
     noway_assert((stack != nullptr) && !stack->empty());
```

Comment:
Do you think that plain asserts are not enough in this case? Looks like this function is hot and can affect throughput.

Reply:
That does seem strange, I'll check. Usually I don't think any kind `x != nullptr` assert is very useful when `x` is immediately deferenced.

Reply:
> Usually I don't think any kind x != nullptr assert is very useful when x is immediately deferenced.

I was thinking about that and I am not sure, noway_assert(false) in release will be caught and the method will be recompiled with min_opts. But will VM handle a SIGSEV?

Reply:
It's JIT's `jitNativeCode` that catches faults and drops to minopts, at least on Windows. I'd expect it also does this on Linux due to the use of `PAL_TRY`.

Reply:
I remember now, I actually checked back then to see what happens if an access violation occurs and the result is not what I expected - the process crashes with a stack overflow exception. I've never been able to figure out, it's difficult to follow what exactly happens in debugger.

In the end I concluded that it's probably safer to just keep `noway_asserts` here. I don't really expect these asserts to ever fire, but that depends on liveness not messing things up and it's kind of difficult to be 100% confident that it won't happen.

So for the time being I think it's best to keep these. Faster code is good but safer code is even better.